schema/spacewalk/common/tables/suseISSAccessToken.sql

```diff
@@ -0,0 +1,26 @@
+--
+-- Copyright (c) 2023 SUSE
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+CREATE TABLE suseISSAccessToken
+(
+    id 	                BIGINT CONSTRAINT suse_isstoken_id_pk PRIMARY KEY
+                          GENERATED ALWAYS AS IDENTITY,
+    token               VARCHAR(1024) NOT NULL,
+    type                VARCHAR(32) NOT NULL
+                            CONSTRAINT suse_isstoken_type_ck
+                            CHECK(type IN ('ISSUED', 'CONSUMED')),
```

Comment:
1) Size limit in this case is not needed and actually hurts performance by always checking for strlen before insert (I know, I know, it's very little).
Limiting VARCHAR size will not save any storage space.

2) Have you considered using enum instead?
```sql
CREATE TYPE suseISSAccessTokenType AS ENUM ('ISSUED', 'CONSUMED');
```

Comparing column sizes then results in:
```
susemanager=# \d test_enums
                   Table "public.test_enums"
  Column   |       Type        | Collation | Nullable | Default
-----------+-------------------+-----------+----------+---------
 type_list | character varying |           | not null |
 type      | type_enum         |           |          |


susemanager=# select pg_column_size(type_list) as list_size, pg_column_size(type) as enum_size from test_enums;
 list_size | enum_size
-----------+-----------
         7 |         4


susemanager=# insert into test_enums values ('ISSUED', 'false value');
ERROR:  invalid input value for enum type_enum: "false value"
```

Reply:
1. I will address it
2. Yes, I'm aware of enum types, but last time we used it, @rjmateus asked us to avoid it because:
     >I'm asking this because normally custom type bring more problems
     >for example, we can only use AWS database (RDS) because is the only one support custom types (since it needs a higher privileged user to create them). In Azure or google cloud we cannot use cloud provider external database

Reply:
Hmm.. I have no experience with managed postgres, but AFAIK to create type you need `CREATE` privilege and we needed it anyway to create new tables.

Reply:
I was looking at the documentation again and it looks like we don't need extra privileges to create an enum type. 
We need extra privileges to create `Composite Types` (like the evr_t we have in our code) which the user must have `USAGE privilege on all attribute types.`. This requirement was only available on AWS / RDS, and not in any other cloud.
For the creation of `Base Types` the user must be a superuser. However, I'm not aware of any of this on our code.

That said, according to the PostgreSQL 16 documentation, the user should not need any extra privileges to use `Enumerated Types` (I didn't test it on AWS/RDS), so we should be safe to move forward with this implementation.

https://www.postgresql.org/docs/16/sql-createtype.html

Reply:
I changed the implementation to use PostgreSQL enums. I also realized that the one we added for PAYG dimension computation was actually never reverted :man_facepalming: